llvm/lib/MC/MCTargetOptionsCommandFlags.cpp

```diff
@@ -108,6 +109,10 @@ llvm::mc::RegisterMCTargetOptionsFlags::RegisterMCTargetOptionsFlags() {
       cl::desc("Emit internal instruction representation to assembly file"));
   MCBINDOPT(ShowMCInst);
 
+  static cl::opt<bool> SSE2AVX(
+      "msse2avx", cl::desc("Convert SSE Instructions to AVX Instructions"));
```

Comment:
This can be "x86-sse2avx"

Reply:
So, we need to pass "-msse2avx" as clang/llvm-mc option, and use "x86-sse2avx" at backend...?
Something like,
```
  static cl::opt<bool> X86Sse2Avx(
      "msse2avx", cl::desc("Convert SSE Instructions to AVX Instructions"));
  MCBINDOPT(X86Sse2Avx);
```

Reply:
@MaskRay requesting for please share your valuable comments on https://github.com/llvm/llvm-project/pull/96860#discussion_r1662228940

Reply:
llvm-mc is an internal tool and its option names are different from clang/gas.... Unlike Clang/GCC, which can define `-m*` options for a specific target, the option names from different targets have a single namespace. It's good to make the target-specific nature clearer.